autofs-5.1.4 - fix error return in do_nfs_mount()

From: Ian Kent <raven@themaw.net>

Fix incorrect error return in modules/parse_amd.c:do_nfs_mount().
---
 CHANGELOG           |    1 +
 modules/parse_amd.c |    2 +-
 2 files changed, 2 insertions(+), 1 deletion(-)

diff --git a/CHANGELOG b/CHANGELOG
index 756ef927..d0cfa19b 100644
--- a/CHANGELOG
+++ b/CHANGELOG
@@ -6,6 +6,7 @@ xx/xx/2018 autofs-5.1.5
 - dont use array for path when not necessary.
 - fix prefix option handling in expand_entry().
 - fix sublink option not set from defaults.
+- fix error return in do_nfs_mount().
 
 19/12/2017 autofs-5.1.4
 - fix spec file url.
diff --git a/modules/parse_amd.c b/modules/parse_amd.c
index 1c962fff..2a5d9a30 100644
--- a/modules/parse_amd.c
+++ b/modules/parse_amd.c
@@ -1099,7 +1099,7 @@ static int do_nfs_mount(struct autofs_point *ap, const char *name,
 	if (strlen(entry->rhost) + strlen(entry->rfs) + 1 > PATH_MAX) {
 		error(ap->logopt, MODPREFIX
 		     "error: rhost + rfs options length is too long");
-		return 0;
+		return 1;
 	}
 
 	strcpy(target, entry->rhost);
